components/policies/staging/base/konflux-rbac/set-tenant-namespace-visibility/set-tenant-namespace-visibility-clusterpolicy.yaml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: set-tenant-namespace-visibility
+  annotations:
+    policies.kyverno.io/description: "Sets the 'konflux-ci.dev/visibility' label on tenant namespaces to 'public' if a RoleBinding for 'system:authenticated' exists, and 'private' if it does not."
+spec:
+  background: true
+  rules:
+  - name: set-visibility-public-if-authenticated-rolebinding-exists
+    match:
+      any:
+      - resources:
+          kinds:
+          - Namespace
+          selector:
+            matchLabels:
+              konflux-ci.dev/type: tenant
+    context:
+      - name: checkAuthenticatedRoleBinding
+        apiCall:
+          urlPath: "/apis/rbac.authorization.k8s.io/v1/namespaces/{{request.object.metadata.name}}/rolebindings"
+          method: GET
+          jmesPath: "items[?subjects[?name=='system:authenticated' && kind=='Group']]"
```

Comment:
we may benefit from using https://kyverno.io/docs/policy-types/cluster-policy/external-data-sources/#global-context here

cc: @amisstea